### PR TITLE
[sailfish-browser] Improve Browser margin handling. Contributes to JB#51962

### PR DESCRIPTION
**Pull request by paveltkv**

### CODE REVIEW

File: src/qtmozembed/declarativewebpage.cpp [hunk 1]
```diff
@@ -263,6 +264,21 @@ void DeclarativeWebPage::setToolbarHeight(qreal toolbarHeight)
     }
 }
 
+bool DeclarativeWebPage::virtualKeyboardOpened() const
+{
+    return m_virtualKeyboardOpened;
+}
+
+void DeclarativeWebPage::setVirtualKeyboardOpened(bool opened)
```
Comment (rainemak) [inline src/qtmozembed/declarativewebpage.cpp:272]: Could you push all this down to the QtMozEmbed.

File: apps/browser/qml/pages/BrowserPage.qml [hunk 2]
```diff
@@ -139,6 +139,9 @@ Page {
             PropertyChanges {
                 target: webView.contentItem
                 virtualKeyboardMargin: virtualKeyboardObserver.panelSize
+                virtualKeyboardHeight: virtualKeyboardObserver.opened
+                                       ? virtualKeyboardObserver.imSize
+                                       : 0
```
Comment (rainemak) [inline apps/browser/qml/pages/BrowserPage.qml:144]:
```
virtualKeyboardHeight: virtualKeyboardObserver.panelSize
```
See when condition. Further, given that virtualKeyboardHeight works is it should you should remove virtualKeyboardMargin (setting margin as well) + cleanup related sendVkbOpenCompositionMetrics from browser side and "embedui:vkbOpenCompositionMetrics" handling from the embedlite-components

File: apps/captiveportal/qml/pages/CaptivePortalPage.qml [hunk 3]
```diff
@@ -105,6 +105,9 @@ Page {
             PropertyChanges {
                 target: webView.contentItem
                 virtualKeyboardMargin: virtualKeyboardObserver.panelSize
+                virtualKeyboardHeight: virtualKeyboardObserver.opened
```
Comment (rainemak) [inline apps/captiveportal/qml/pages/CaptivePortalPage.qml:108]: Same as above

**Review by rainemak — CHANGES_REQUESTED** [review]
Please cleanup virtualKeyboardMargin and related code that will obsoleted once it is removed.

**Review by llewelld — APPROVED** [review]
This looks good to me and works nicely. It's also an important improvement for the browser, so really nice to have.

I think at this stage it might be better to combine #733 and #741 into one (possibly two) commits. Otherwise you have code added/changed that gets immediately removed.

File: apps/captiveportal/qml/pages/CaptivePortalPage.qml [hunk 4]
```diff
@@ -104,7 +104,7 @@ Page {
             when: virtualKeyboardObserver.opened && webView.enabled
             PropertyChanges {
                 target: webView.contentItem
-                virtualKeyboardMargin: virtualKeyboardObserver.panelSize
+                virtualKeyboardHeight: virtualKeyboardObserver.imSize
```
Comment (rainemak) [inline apps/captiveportal/qml/pages/CaptivePortalPage.qml:107]: Need to check in more detail but we should not use the value that is animated -- only one single update and that's it.

File: apps/qtmozembed/declarativewebpage.cpp [hunk 5]
```diff
@@ -432,40 +402,6 @@ void DeclarativeWebPage::resetViewMargins()
     setMargins(margins);
 }
```
Comment (rainemak) [inline apps/qtmozembed/declarativewebpage.cpp:404]: This doesn't build now...

See DeclarativeWebPage::updateViewMargins() and m_virtualKeyboardMargin. Just change to use virtualKeyboardHeight().

**Review by rainemak — CHANGES_REQUESTED** [review]
See one build error. After that is fixed, let's get this one merged.

File: apps/captiveportal/qml/pages/CaptivePortalPage.qml (hunk 4 above)
Comment (rainemak) [reply apps/captiveportal/qml/pages/CaptivePortalPage.qml:107]: This is good.

**Comment by paveltkv** [comment]
Rebased, combined with #733 and removed margin change animation.

**Review by llewelld — APPROVED** [review]
This looks good to me, thank you Pavel. In testing it worked nicely, no glitches or obvious performance issues. The code looks good too. Re-approving

**Comment by rainemak** [comment]
> This looks good to me and works nicely. It's also an important improvement for the browser, so really nice to have.
> 
I do agree with this comment. Also worth saying that more testing we get for this, sooner we find the issues and have them fixed.

**Review by rainemak — APPROVED** [review]
Looks good now and works well! Thanks @paveltkv

**Review by okodron — APPROVED** [review]

**Pull request merged**